### PR TITLE
Tools: Add  UIO driver support to PAC N3000 ethernet tools

### PR DESCRIPTION
**Pull request by anandaravuri**
Upstream DFL VFIO MDEV driver patch was rejected by Linux, DFL driver supports UIO driver instead of VFIO MDEV.
UIO driver can used to access  N3000 ethernet DFH.

eth_group_uio.sh: Enables / Disables  PAC N3000 card ethernet group.
eth_group Module: this pybind11 module  opens eth group DFH and reads/writes CSRs.

Updated common.py, fecmode.py, fpgamac.py, fpgastats.py, fvlbypass.py, mactest.py tools to support UIO Driver

Signed-off-by: anandaravuri <ananda.ravuri@intel.com>

### CODE REVIEW

File: scripts/eth_group_uio.sh [hunk 1]
```diff
@@ -1,7 +1,7 @@
 #!/bin/bash
```
Comment (tswhison) [inline scripts/eth_group_uio.sh:1]: This file is a duplicate of the opaeuio.py script, which is part of opae.admin.  I think it can be safely removed.

Comment (anandaravuri) [reply scripts/eth_group_uio.sh]: remove eth_group_uio.sh ,  use opaeuio.py script

opaeuio dfl_dev.10 -i -u lab -g lab 
opaeuio dfl_dev.11 -i -u lab -g lab

Comment (anandaravuri) [reply scripts/eth_group_uio.sh]: fixed

File: tools/extra/fpgadiag/eth_group.cpp [hunk 2]
```diff
@@ -45,7 +45,7 @@
 #define ETH_GROUP_MAC		2
 #define ETH_GROUP_ETHER		3
 
-#define DEV_VFIO_PATH  "/dev/vfio/vfio"
+//#define DEV_VFIO_PATH  "/dev/vfio/vfio"
```
Comment (r-rojo) [inline tools/extra/fpgadiag/eth_group.cpp:48]: remove dead code

File: tools/extra/fpgadiag/eth_group.cpp [hunk 3]
```diff
@@ -54,140 +54,49 @@
 
 
 // open eth group
-int eth_group::eth_group_open(int vfio_id, std::string fpga_mdev_str)
+int eth_group::eth_group_open(std::string fpga_uid_str)
```
Comment (r-rojo) [inline tools/extra/fpgadiag/eth_group.cpp:57]: Can fpga_uid_str be a const std::string&?

File: tools/extra/fpgadiag/eth_group.cpp [hunk 4]
```diff
@@ -54,140 +54,49 @@
...
 
-	// Add the group to the container
-	ioctl(group, VFIO_GROUP_SET_CONTAINER, &container);
+	mmap_ptr = mem;
+	ptr_ = (uint64_t*)mem;
 
-	// Enable the IOMMU model
-	ioctl(container, VFIO_SET_IOMMU, VFIO_TYPE1_IOMMU);
+	eth_info.csr = *(ptr_ + 1);
+	eth_dfh.csr = *(ptr_);
```
Comment (r-rojo) [inline tools/extra/fpgadiag/eth_group.cpp:77]: Is there any identifying information in the DFH or the eth_info (like a GUID) that can be used to verifythe registers before poking them?

File: tools/extra/fpgadiag/eth_group.cpp [hunk 5]
```diff
@@ -54,140 +54,49 @@
...
 
 // reset mac
 bool eth_group::mac_reset()
 {
-	uint32_t i;
-	uint32_t value;
-	int ret = 0;
+	uint32_t i       = 0;
+	uint32_t value   = 0;
+	int ret           = 0;
```
Comment (r-rojo) [inline tools/extra/fpgadiag/eth_group.cpp:99]: I don't understand why we need mac_value. Can this be simplified to something like this?

```c++
// reset mac
bool eth_group::mac_reset()
{
  for (uint32_t i = 0; i < phy_num; i++) {
    if (read_reg(ETH_GROUP_MAC, i, 0, MAC_CONFIG) == ETH_GROUP_RET_VALUE))
        return false;
    if (write_reg(ETH_GROUP_MAC, i, 0, MAC_CONFIG, 0x0))
      return false;
  return true;
}
```

File: tools/extra/fpgadiag/eth_group.cpp (hunk 2 above)
Comment (anandaravuri) [reply tools/extra/fpgadiag/eth_group.cpp]: Removed Dead code, fixed

File: tools/extra/fpgadiag/eth_group.cpp (hunk 3 above)
Comment (anandaravuri) [reply tools/extra/fpgadiag/eth_group.cpp]: Changed input to open to  const std::string&, fixed

File: tools/extra/fpgadiag/eth_group.cpp (hunk 5 above)
Comment (anandaravuri) [reply tools/extra/fpgadiag/eth_group.cpp]: Removed  mac_value variable and simplified code

File: tools/extra/fpgadiag/eth_group.cpp (hunk 4 above)
Comment (anandaravuri) [reply tools/extra/fpgadiag/eth_group.cpp:74]: Check with Feature ID, doesn't support GUID.
	// Check ETH group FeatureID
	if (eth_dfh.id != ETH_GROUP_FEATUREID) {
		printf("Worng Eth group Feature ID \n");
		return -1;
	}

**Pull request closed without merging**